tests/integration.rs

```diff
@@ -0,0 +1,195 @@
+use serial_test::serial;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use bevy::MinimalPlugins;
+use bevy::input::{ButtonState, Input, InputPlugin};
+use bevy::input::keyboard::KeyboardInput;
+use bevy::prelude::*;
+use bevy::utils::HashMap;
+use ggrs::{Config, SessionBuilder, PlayerType, UdpNonBlockingSocket, PlayerHandle, P2PSession};
+use bevy_ggrs::{GgrsPlugin, AdvanceFrame, PlayerInputs, Rollback, Session, LocalInputResource};
+use bytemuck::{Pod, Zeroable};
+
+#[test]
+#[serial]
+fn it_runs_advance_frame_schedule_systems() -> Result<(), Box<dyn std::error::Error>> {
+    let (player1, player2) = create_players();
+    let session1 = start_session(&player1, &player2)?;
+    let mut app1 = create_app::<GgrsConfig>(session1);
+    let session2 = start_session(&player2, &player1)?;
+    let mut app2 = create_app::<GgrsConfig>(session2);
+
+    let inputs1 = HashMap::from([
+        (player1.handle, BoxInput { inp: 0 }),
+    ]);
+    let inputs_resource = LocalInputResource::<GgrsConfig> {
+        inputs: inputs1,
+    };
+    app1.insert_resource(inputs_resource);
+    for _ in 0..50 {
+        app1.update();
```

Comment:
Does bevy_ggrs read from the Time resource? If so, I don't think this will actually advance the game 50 frames, probably just one frame.

I tried making some test for bevy_xpbd based on these ones, and I had to manually update the time resource, see https://github.com/Jondolf/bevy_xpbd/pull/18/files#diff-a0262fea108af44a98cc49e5eb72641963dd816513f2d0a22eb738fcfed55ebe

Reply:
You are correct, it doesn't advance the game by 50 frames. So yeah in this case probably a few frames, depending on CI machine. To my understanding `app.update()` advances the advances systems by one cycle (whatever that means).

If you want to advance by frame/time I think a great example is taken from this ggrs integration test:
https://github.com/gschup/bevy_ggrs/blob/95421952b83442cd103bb42301b5d2f1749a2dfa/tests/entity_maping.rs#L103

The hard part for me is that the time/cycles it needs to actually connect both apps to each peer varies between machines. So splitting the connection part and the frame advance systems by time can be tricky to run in a CI machine.

I really like what you did here: https://github.com/Jondolf/bevy_xpbd/blob/11499efaf5b039ba356f029d11f976449842cca2/src/tests.rs#L33

This is very useful for testing your system on a frame by frame scenario. 

But to be honest if I would be testing a system in a frame by frame scenario I would probably isolate the FrameCount and the system under test by just stubbing FrameCount and testing the system in a more unit test fashion.

Reply:
Are the systems here demanding enough to take more than 16ms of cpu-time, though?

Reply:
I don't know. 
Why you ask?

Reply:
Just that I think ggrs update frequency is set to 60 fps, so if you don't run for at least 16 ms, you'll never get to the next frame, and you might not actually test what you think you're testing.